Brokerages/Backtesting/BasicOptionAssignmentSimulation.cs

```diff
@@ -136,8 +138,9 @@ public void SimulateMarketConditions(IBrokerage brokerage, IAlgorithm algorithm)
 
             algorithm.Securities
                 // we take only options that expire soon
-                .Where(x => (x.Key.ID.SecurityType == SecurityType.Option || x.Key.ID.SecurityType == SecurityType.FutureOption) &&
-                            x.Key.ID.Date - algorithm.UtcTime <= _priorExpiration)
+                .Where(x => x.Key.ID.SecurityType.IsOption() &&
+                            ((x.Key.ID.OptionStyle == OptionStyle.American && x.Key.ID.Date - algorithm.UtcTime <= _priorExpiration) ||
+                            (x.Key.ID.OptionStyle == OptionStyle.European && x.Key.ID.Date.Date == algorithm.UtcTime.Date)))
```

Comment:
Hm here we use UTC and above we use algorithm time, behavior is master though 🤔🐟